### PR TITLE
Update Invoke-GraphQLQuery.ps1

### PR DESCRIPTION
**Pull request by luchsamapparat**
Thanks for providing this module!

We're using it to import data from a UTF-8 encoded CSV file via a GraphQL API. This script is executed in PowerShell Core, which also uses UTF-8 by default (afaik). It seems, that special characters like German umlauts change to gibberish when [the request is invoked](https://github.com/anthonyg-1/PSGraphQL/blob/main/PSGraphQL/Functions/Invoke-GraphQLQuery.ps1#L346) using `Invoke-RestMethod` without any charset information.

We were able to solve this issue by explicitly defining UTF-8 as charset as part of the `Content-Type` header.

Another solution might be to add an additional parameter to `Invoke-GraphQLQuery` to define the charset. But I'm not sure if that's necessary.

### CODE REVIEW

**Comment by anthonyg-1** [comment]
Hi. Before I approve I'd like this tested on Linux. Can I assume you're running this on Windows currently? Is there any chance you have a Linux or MacOS machine you can test this from? I'm reading that setting that value in Linux and/or MacOS may cause issues: https://github.com/itglue/powershellwrapper/issues/12

**Comment by anthonyg-1** [comment]
Just tested it and looks good. Approved.

**Comment by luchsamapparat** [comment]
Yes, you're right. I only tested it on a Windows machine. By I will try it on a Mac next week.

On second thought, I'm not even sure if the proposed solution is such a good idea. Depending on where your data comes from, it might be UTF-8, ISO-8859-1, or something else. So maybe a better/more flexible approach would be keep the current charset as default, and allow defining another one via param.

**Comment by luchsamapparat** [comment]
Now, you were faster than I was answering your first post :)

**Comment by anthonyg-1** [comment]
Hmm...ok will rollback then. Any chance you want to add that new parameter? I would suggest we make it optional with a defaulted value.

**Comment by luchsamapparat** [comment]
Can this module also be used in a Windows PowerShell context? If that's the case, it could cause trouble for existing users, as Windows PowerShell is windows-1252 by default.

**Comment by luchsamapparat** [comment]
Yes, no problem. I'll send you an updated PR tomorrow

**Comment by anthonyg-1** [comment]
Awesome much appreciated! Please test against Windows PowerShell as well as on Linux or MacOS. Also please update help for the parameter you wish to add.

**Comment by luchsamapparat** [comment]
I did some testing to verify my assumptions. I built a small webserver that logs the actual bytes it receives and then used the same PowerShell command in Windows PowerShell, PowerShell Core on Windows and PowerShell Core on Mac to send a request to that server:

```ps1
Invoke-WebRequest -Uri http://localhost:3000/ -Method Post -Body "{ `"foo`": `"ä`" }" -ContentType "application/json"
```

On **all platforms** (PowerShell Core on Win/Mac OS, Windows Powershell) I received these bytes as request body:

```
7b 20 22 66 6f 6f 22 3a 20 22 e4 22 20 7d
```

[DenCode tells us](https://dencode.com/string/hex?v=7b%2020%2022%2066%206f%206f%2022%203a%2020%2022%20e4%2022%2020%207d&oe=ISO-8859-1&nl=crlf), that this is **ISO-8859-1**.

Then I created a UTF-8 encoded JSON file (similar to the CSV files I was trying to import) with the same JSON data:

```
{ "foo": "ä" }
```

And then sent the contents of that file as request body using these commands:

```ps1
# without -Encoding parameter
Invoke-WebRequest -Uri http://localhost:3000/ -Method Post -Body (Get-Content ./utf8-data.json) -ContentType "application/json"

# with -Encoding parameter
Invoke-WebRequest -Uri http://localhost:3000/ -Method Post -Body (Get-Content ./utf8-data.json -Encoding utf8) -ContentType "application/json"
```


On **PowerShell Core (Windows)** and **PowerShell Core (Mac OS)** I receive the same **ISO-8859-1** bytes as above:

```
7b 20 22 66 6f 6f 22 3a 20 22 e4 22 20 7d
```

On **Windows PowerShell** I get the same bytes as above when using the `-Encoding utf8` parameter for `Get-Content`. Otherwise, I get this:

```
7b 20 22 66 6f 6f 22 3a 20 22 c3 a4 22 20 7d
```
[DenCode confirms](https://dencode.com/string/hex?v=7b%2020%2022%2066%206f%206f%2022%203a%2020%2022%20c3%20a4%2022%2020%207d&oe=UTF-8&nl=crlf), that this is UTF-8.

And that's probably reasonable. If I don't tell Windows PowerShell the encoding of a file, it assumes it is the default charset (Windows-1252 - which for the sake of this discussion is more or less equivalent to ISO-8859-1) and skips any conversion.

---

To conclude all this, I modified our CSV import script to send its requests to my test server and the result matches everything I observed above: I receive ISO-8859-1 encoded data.

In other words, the problem isn't this module, it's somewhere in our server implementation. 🤦‍♂️

Sorry, for all the confusion.
